networkx/algorithms/matching.py

```diff
@@ -92,6 +92,13 @@ def is_matching(G, matching):
     """
     if isinstance(matching, dict):
         matching = matching_dict_to_set(matching)
+
+    for edge in matching:
+        if len(edge) != 2:
```

Comment:
When does this situation arise; where `len(edge) != 2`?

Reply:
See the first revision in this PR (specifically the test failure: https://github.com/networkx/networkx/runs/1974404252#step:5:487) in that test one of the values of `edge` was `frozenset{1}` when there was a self loop. I figured since a self loop was an invalid matching it was better to just handle that edge case like this.

Reply:
This also provides a check for 3-tuple representations of an edge (either including an edge key from a multigraph or a data dict using `G.edges.data()`). 

But that is a double-edge sword because it could hide a misspecified input `matching`. Certainly it is true that a mis-specified matching is not a matching. So, returning `False` makes sense, but potentially hides a bug in the user's code.

@mtreinish how did the input matching become a collection of `frozenset`s to produce `frozenset({1})`. Was the code that converted a self-loop edge from a `(n, n)` 2-tuple to a `frozenset({n})` part of NetworkX or something that you created as a test or application?

Reply:
It was coming from an existing networkx test: https://github.com/networkx/networkx/blob/master/networkx/algorithms/tests/test_matching.py#L488-L496 that's why I was surprised when things failed after I pushed the PR because it seemed straightforward. The cause of this is this line: https://github.com/networkx/networkx/blob/master/networkx/algorithms/matching.py#L137 in `is_maximal_matching` which calls `is_matching` internally.